### ISSUE TITLE
Start kernel on correct local path

### ISSUE THREAD

**Issue by diocas** [sciencemesh/cs3api4lab#7]
The lab extension will connect directly using the CS3APIs. But we also need to mount the file system using the webdav (or EOS) FUSE, to allow local access from the kernel.
We need to ensure that opening the notebook starts the kernel in the respective mounted path (which means we might need to translated the paths from the CS3 API to the local FUSE mounted filesystem).

**Comment by JarCz** [comment]
In this case, we need to consider and resolve the problems:
- each system have a different filesystem layout
- automatic change the path to mount point 
1. Find the path to the mounting point in the filesystem:
- set the path to the mounting point in the config file
- automatic search in the file system - at startup plugin create "ID file" in the IOP and search in the file system; long search time in the large filesystem or network filesystems and freeze Jupyter in the startup process
- ?
2. Setup the path in the kernel and env:
- set environment value "c.NotebookApp.notebook_dir"
- automatically add the "os.chdir(path)" to all opened python notebooks to change the working directory
- add the plugin to the kernel to translate paths - is it possible?
- ?

**Comment by diocas** [comment]
For knowing the path in the local mountpoint, I think we can start with a basic config (we might need something more configurable, but for now should be fine this way).
For example, for resolving personal files + normal shares I would suggest we know how to translate a user home directory into a local path. If you set an option in our contentsmanager like `local_home = "/eos/user/{username[0]}/{username}/", then you know that any file that belongs to a user needs to be inside that one.

For setting the path in the kernel, I'm absolutely sure we should not add kernel extensions. As you said, I would set `notebook_dir` to some base path (we could take the first non-variable part of `local_home`, which in this example would be `/eos/user`, or just put `/`). Then in the file metadata you return `path` built by removing the subpath you added as `notebook_dir`. 
For example, if I have a file `myfolder/mynotebook.ipynb` (on my home), then `path` would be `d/diogo/myfolder/mynotebook.ipynb` (if you have `notebook_dir = "/eos/user"`. This would result in my kernel starting in `/eos/user/d/diogo/myfolder/`.

**Comment by JarCz** [comment]
Ok, it's clear:
* we don't modify the kernel, we work only on the backend
* add value in the config and parse this value - username, etc
* modified the value like c.NotebookApp.notebook_dir from the backend

@diocas can you prepare more test examples from the CERN environment or you are used only this mapping type? 
for example, the path from IOP for user Diogo `/home/myfolder/notebok.py` it's mapped in local to `/eos/user/d/diogo/myfolder/notebok.py`

**Comment by diocas** [comment]
@piotrWichlinskiSoftwaremind @mateusz-garlacz @dagl
Good news guys, as we discussed I've looked a bit more into this and it's as easy as I said (maybe even more).

So, when you open a notebook, Jupyter sends the `path` of the notebook to the backend (session manager) to start the kernel. But the session manager will call the method `get_kernel_path` from `contentsmanager` which returns the directory that will be set as the working directory.
So we just need to implement this method!

I said in previous comments that we can pass a configuration parameter to do the translation, but maybe we can make it a python method/callable instead of a string (with current username and the path as parameters). This gives more freedom to configure. But as you prefer, we can start simple.
